sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusRetryPolicy.cs

```diff
@@ -28,10 +28,11 @@ public abstract class ServiceBusRetryPolicy
     {
```

Comment:
I don't know that we have test coverage for this functionality at all.   Should we consider adding some in the [BasicRetryPolicy tests](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/tests/Core/BasicRetryPolicyTests.cs) or roll a test class dedicated for the base functionality?

Reply:
This would probably require to make the `ServerBusyBaseSleepTime`, `IsServerBusy` and `ServerBusyExceptionMessage` and internal property. I can push a proposal.

Reply:
Pushed a few tests to get started so that you can see whether this goes into the right direction

Reply:
Approach looks good to me.